src/zmalloc.c

```diff
@@ -87,10 +88,31 @@ void zlibc_free(void *ptr) {
 #define dallocx(ptr, flags) je_dallocx(ptr, flags)
 #endif
 
-#define update_zmalloc_stat_alloc(__n) atomic_fetch_add_explicit(&used_memory, (__n), memory_order_relaxed)
-#define update_zmalloc_stat_free(__n) atomic_fetch_sub_explicit(&used_memory, (__n), memory_order_relaxed)
+#define update_zmalloc_stat_alloc(__n)                                                                                 \
+    do {                                                                                                               \
+        if (unlikely(thread_index == -1)) zmalloc_register_thread_index();                                             \
+        used_memory_thread[thread_index] += (__n);                                                                     \
+    } while (0)
```

Comment:
Minor: We could use a static inline function for this instead of a macro.

Reply:
Done.